### PR TITLE
[Core] Provide a way to check the active scm host

### PR DESCRIPTION
**Pull request by k0nserv**
Adds `scm_provider` to the DSL which will let users and plugins check
what the active scm_provider is. Example

```ruby
  if scm_provider == :gitlab do
    message("Running under gitlab")
  end
```

### CODE REVIEW

File: lib/danger/danger_core/dangerfile_dsl.rb [hunk 1]
```diff
@@ -13,6 +13,19 @@ module DSL
...
+      # @return [Symbol] The name of the SCM Provider used for the active repository.
+      def scm_provider
+        if env.request_source && env.request_source.class == Danger::RequestSources::GitHub
+          :github
+        elsif env.request_source && env.request_source.class == Danger::RequestSources::GitLab
+          :gitlab
+        else
+          :unknown
+        end
+      end
```
Comment (orta) [inline lib/danger/danger_core/dangerfile_dsl.rb]: this this might be best on the `danger` plugin, it'll then get documented in the reference 👍

**Comment by JuanitoFatas** [comment]
We also support [bitbucket server](https://github.com/danger/danger/blob/master/CHANGELOG.md#320)? :)

**Comment by k0nserv** [comment]
Is the bitbucket support merged?

File: lib/danger/danger_core/dangerfile_dsl.rb (hunk 1 above)
Comment (k0nserv) [inline lib/danger/danger_core/dangerfile_dsl.rb]: Will do :)

**Comment by JuanitoFatas** [comment]
> Is the bitbucket support merged?

Yea merged in https://github.com/danger/danger/pull/496, shipped yesterday in 3.2.0. :)

**Comment by k0nserv** [comment]
👌 will update

**Comment by orta** [comment]
10/10 👌 
awesome

**Pull request merged**